mission/waypoint_manager/src/waypoint_manager.cpp

```diff
@@ -0,0 +1,288 @@
+#include "waypoint_manager/waypoint_manager.hpp"
+#include <functional>
+#include <future>
+#include <memory>
+#include <thread>
+
+using namespace std::placeholders;
+
+WaypointManagerNode::WaypointManagerNode() : Node("waypoint_manager_node") {
+    server_cb_group_ =
+        this->create_callback_group(rclcpp::CallbackGroupType::Reentrant);
+    client_cb_group_ =
+        this->create_callback_group(rclcpp::CallbackGroupType::Reentrant);
+
+    setup_action_server();
+    setup_action_clients();
+
+    running_ = true;
+    worker_thread_ =
+        std::thread(&WaypointManagerNode::process_waypoints_thread, this);
+
+    spdlog::info("Waypoint Manager Node initialized");
+}
+
+WaypointManagerNode::~WaypointManagerNode() {
+    {
+        std::lock_guard<std::mutex> lock(queue_mutex_);
+        running_ = false;
+    }
```

Comment:
Could let running_ be a `std::atomic<bool>` to avoid having to grab the lock here?